iotdb-core/datanode/src/main/java/org/apache/iotdb/db/queryengine/plan/execution/config/executor/ClusterConfigTaskExecutor.java

```diff
@@ -2790,6 +2795,119 @@ public SettableFuture<ConfigTaskResult> showDatabases(ShowDB showDB) {
     return future;
   }
 
+  @Override
+  public SettableFuture<ConfigTaskResult> showCluster(ShowCluster showCluster) {
+    SettableFuture<ConfigTaskResult> future = SettableFuture.create();
+    TShowClusterResp showClusterResp = new TShowClusterResp();
+    try (ConfigNodeClient client =
+        CONFIG_NODE_CLIENT_MANAGER.borrowClient(ConfigNodeInfo.CONFIG_REGION_ID)) {
+      showClusterResp = client.showCluster();
+    } catch (ClientManagerException | TException e) {
+      if (showClusterResp.getConfigNodeList() == null) {
+        future.setException(new TException(MSG_RECONNECTION_FAIL));
+      } else {
+        future.setException(e);
+      }
+      return future;
+    }
+    // build TSBlock
+    if (showCluster.getDetails().orElse(false)) {
+      ShowClusterDetailsTask.buildTSBlock(showClusterResp, future);
+    } else {
+      org.apache.iotdb.db.queryengine.plan.execution.config.metadata.relational.ShowClusterTask
+          .buildTsBlock(showClusterResp, future);
+    }
+
+    return future;
+  }
+
+  @Override
+  public SettableFuture<ConfigTaskResult> showRegions(ShowRegions showRegions) {
+    SettableFuture<ConfigTaskResult> future = SettableFuture.create();
+    TShowRegionResp showRegionResp = new TShowRegionResp();
+    TShowRegionReq showRegionReq = new TShowRegionReq();
+    showRegionReq.setConsensusGroupType(showRegions.getRegionType());
+    if ((showRegions.getStorageGroups() == null) || showRegions.getStorageGroups().isEmpty()) {
+      showRegionReq.setDatabases(null);
+    } else {
+      showRegionReq.setDatabases(
+          showRegions.getStorageGroups().stream()
+              .map(PartialPath::getFullPath)
+              .collect(Collectors.toList()));
+    }
+    try (ConfigNodeClient client =
+        CONFIG_NODE_CLIENT_MANAGER.borrowClient(ConfigNodeInfo.CONFIG_REGION_ID)) {
+      showRegionResp = client.showRegion(showRegionReq);
+      if (showRegionResp.getStatus().getCode() != TSStatusCode.SUCCESS_STATUS.getStatusCode()) {
+        future.setException(
+            new IoTDBException(
+                showRegionResp.getStatus().message, showRegionResp.getStatus().code));
+        return future;
+      }
+    } catch (ClientManagerException | TException e) {
+      future.setException(e);
+    }
+
+    // filter the regions by nodeId
+    // TODO: This part currently will remain empty as we're ot implementing any other filtering
+    // methods right now.
+    if ((showRegions.getNodeIds() != null) && !showRegions.getNodeIds().isEmpty()) {
```

Comment:
redundant ()?

Reply:
Well not really ... as if getNodeIds returns null, the "isEmpty" call will cause an NPE, but it could also be an empty collection.

Reply:
I mean we could use
`if (showRegions.getNodeIds() != null && !showRegions.getNodeIds().isEmpty()) {`

Reply:
I usually add them additionally in all expressions as I've seen in the past not everyone is familiar with the priorities of computational operations. () don't cost anything, but they make it impossible to mis-interpret.